pkg/registry/resource/resourceslice/strategy.go

```diff
@@ -60,8 +61,17 @@ func (resourceSliceStrategy) Validate(ctx context.Context, obj runtime.Object) f
 	return validation.ValidateResourceSlice(slice)
 }
 
+// WarningsOnCreate returns warnings for the creation of the given object.
 func (resourceSliceStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string {
-	return nil
+	newResourceSlice := obj.(*resource.ResourceSlice)
+	var warnings []string
+
+	if newResourceSlice.Spec.Driver != strings.ToLower(newResourceSlice.Spec.Driver) {
+		warnings = append(warnings,
+			fmt.Sprintf("spec.driver: driver names should be lowercase; %q contains uppercase characters", newResourceSlice.Spec.Driver))
```

Comment:
We should also document this for DRA driver admins:
- https://github.com/kubernetes/kubernetes/blob/0b6dba8eb028bb0589076891458977cb88afefbb/staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go#L207-L208
- https://github.com/kubernetes/kubernetes/blob/0b6dba8eb028bb0589076891458977cb88afefbb/staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go#L196-L197
- https://github.com/kubernetes/kubernetes/blob/0b6dba8eb028bb0589076891458977cb88afefbb/staging/src/k8s.io/api/resource/v1/types.go#L103-L104 (and all other API versions)

The explanation in all places can be "Must be a DNS subdomain and should end with a DNS domain owned by the vendor of the driver. It should use only lower case characters."